src/commands/new.ts

```diff
@@ -12,84 +12,291 @@ import {
+    let targetPath = useDefault(options.targetPath) ? defaultTargetPath : options.targetPath
+    if (targetPath === undefined) {
+      const targetPathResponse = await prompt.ask({
+        type: "input",
+        name: "targetPath",
+        message: "Where do you want to start your project?",
+        initial: defaultTargetPath,
+      })
+
+      targetPath = targetPathResponse.targetPath
+    }
+
+    // #endregion
+
+    // #region Overwrite
+    // if they pass in --overwrite, remove existing directory otherwise throw if exists
+    const defaultOverwrite = false
+    let overwrite = useDefault(options.overwrite) ? defaultOverwrite : options.overwrite
+    if (exists(targetPath)) {
+      if (overwrite === undefined) {
+        overwrite = await prompt.confirm(
+          `${targetPath} already exists. Do you want to overwrite it?`,
+          false,
+        )
+      }
+
+      if (overwrite === false) {
+        const alreadyExists = `Error: There's already a folder at ${targetPath}. To force overwriting that folder, run with --overwrite or say yes.`
+        p()
+        p(yellow(alreadyExists))
+        process.exit(1)
+      }
+
+      log(`Removing existing project ${projectName}`)
+      remove(projectName)
+    }
+    // #endregion
+
+    // #region Prompt Git Option
+    const defaultGit = true
+    let git = useDefault(options.git) ? defaultGit : options.git
+
+    if (git === undefined) {
+      git = await prompt.confirm("Do you want to initialize a git repository?", true)
+    }
+    // #endregion
 
+    // #region Packager
     // check if a packager is provided, or detect one
     // we pass in expo because we can't use pnpm if we're using expo
-    const packagerName = parameters.options.packager || packager.detectPackager()
+
+    const availablePackagers = packager.availablePackagers()
+    const defaultPackagerName = availablePackagers.includes("yarn") ? "yarn" : "npm"
+    let packagerName = useDefault(options.packager) ? defaultPackagerName : options.packager
+
+    const validatePackagerName = (input: unknown): input is PackagerName =>
+      typeof input === "string" && ["npm", "yarn", "pnpm"].includes(input)
+
+    if (packagerName !== undefined && validatePackagerName(packagerName) === false) {
+      p()
+      p(yellow(`Error: Invalid packager: "${packagerName}". Valid packagers are npm, yarn, pnpm.`))
+      process.exit(1)
+    }
+
+    if (packagerName !== undefined && availablePackagers.includes(packagerName) === false) {
+      p()
+      p(yellow(`Error: selected "${packagerName}" but packager was not available on system`))
+      process.exit(1)
+    }
+
+    if (packagerName === undefined) {
+      const initial = availablePackagers.findIndex((p) => p === defaultPackagerName)
+      const NOT_FOUND = -1
+
+      if (initial === NOT_FOUND) {
+        p()
+        p(yellow(`Error: Default packager "${defaultPackagerName}" was not available on system`))
+        process.exit(1)
+      }
```

Comment:
This should never happen, but I felt it was important to document since it wasn't intuitive to me that `findIndex` returns -1 if not found

Reply:
Fair enough! JS is weird sometimes.